packages/utils/typescript/lib/generators/schemas/index.js

```diff
@@ -0,0 +1,173 @@
+'use strict';
+
+const path = require('path');
+
+const ts = require('typescript');
+const { factory } = require('typescript');
+
+const fp = require('lodash/fp');
+const fse = require('fs-extra');
+const prettier = require('prettier');
+const chalk = require('chalk');
+const CLITable = require('cli-table3');
+
+const { generateImportDefinition } = require('./imports');
+const { generateSchemaDefinition } = require('./schema');
+const { generateGlobalDefinition } = require('./global');
+const {
+  getAllStrapiSchemas,
+  getSchemaInterfaceName,
+  getSchemaModelType,
+  getDefinitionAttributesCount,
+} = require('./utils');
+
+const DEFAULT_OUT_FILENAME = 'schemas.d.ts';
+
+/**
+ * Generate type definitions for Strapi schemas
+ *
+ * @param {object} options
+ * @param {Strapi} options.strapi
+ * @param {{ distDir: string; appDir: string; }} options.dirs
+ * @param {string} [options.outDir]
+ * @param {string} [options.file]
+ * @param {boolean} [options.debug]
+ */
+const generateSchemasDefinitions = async (options = {}) => {
+  const { strapi, outDir = process.cwd(), file = DEFAULT_OUT_FILENAME, debug = false } = options;
+
+  const schemas = getAllStrapiSchemas(strapi);
+
+  const schemasDefinitions = Object.values(schemas).map(schema => ({
+    schema,
+    definition: generateSchemaDefinition(schema),
+  }));
+
+  const allDefinitions = [
+    // Imports
+    generateImportDefinition(),
+
+    // Add a newline after the import statement
+    factory.createIdentifier('\n'),
+
+    // Schemas
+    ...schemasDefinitions.reduce(
+      (acc, def) => [
```

Comment:
I don't think this has much an impact compared to the generator themselves but just in case, for perf reason, it might be worth pushing to `acc` instead of concatenating